packages/cloudflare/src/cli/config.ts

```diff
@@ -49,8 +49,9 @@ export type Config = {
  * @returns The configuration, see `Config`
  */
 export function getConfig(projectOpts: ProjectOptions): Config {
+  const sourceDirDotNext = join(projectOpts.sourceDir, ".next");
   const dotNext = join(projectOpts.outputDir, ".next");
-  const appPath = getNextjsApplicationPath(dotNext).replace(/\/$/, "");
+  const appPath = getNextjsApplicationPath(sourceDirDotNext).replace(/\/$/, "");
```

Comment:
The reason why we copy `.next` to the output is to keep the source `.next` unchanged.
That way we can skip building next during development and we are still sure to have a pristine `.next`.
Let's say you are updating a patch modifying files in the `.next` folder, you would have to build the next app every time you change it if it was working on the source `.next`

Reply:
The CLI currently fails to build in skip-build mode if no .next folder exists, so i guess that needs fixing as well if it's expected to work entirely independently of the .next directory.

Reply:
Would you be okay with this doing if the dotnext dir exists, use that, otherwise use the source dir?

Reply:
The workflow is: first do a full build (create the `.next`) and then you can iterate without having to rebuild next.

> so i guess that needs fixing as well if it's expected to work entirely independently of the .next directory.

It's mostly for dev usage so I don't think this needs to be perfect. But sure we could still build in skip build if there is no `.next` folder. Honestly not a high priority - but fell free to improve that.

> Would you be okay with this doing if the dotnext dir exists, use that, otherwise use the source dir?

Not sure to understand what you want to do here?

Reply:
> not a high priority

If you want to contribute things that will be needed soon and have a higher impact, you can take a look at the [project board](https://github.com/orgs/opennextjs/projects/1).

I am working on integrating this adapter with opennext-aws. I will hopefully have midlleware + server running by the end of the week.

Next we will work on integrating the cache and the image optimization. If you feel like looking at one of this task it would really help. We need to make those run on the cloudflare infra.

You can ping me on Discord if you have questions.

Thanks!

Reply:
> Not sure to understand what you want to do here?

i was going to enable the config to be creation without depending on the output directory existing. I'll remove that change. This PR is just improving separation of source / output / internal paths in the config now. Can close this if you don't think that's valuable though.

> Next we will work on integrating the cache and the image optimization. If you feel like looking at one of this task it would really help. We need to make those run on the cloudflare infra.

I would quite like to help get some of the cache stuff moving but it's not clear how the merge with the aws adaptor code is supposed to happen - new repos? new monorepo? copy paste?

It talks about multiple possibilities being explored already with no information or context about those explorations or the outcomes, so unsure what the current state is

Reply:
> I would quite like to help get some of the cache stuff moving but it's not clear how the merge with the aws adaptor code is supposed to happen - new repos? new monorepo? copy paste?

Great to hear that!

Ultimately we will probably want a monorepo - this is something we will have to discuss with all the contributors.

For now the idea would be to have a `build.ts` that is specific to cloudflare but can still make use of the code in `was`. I have started to submit PRs to `aws` so that it's easier to re-use the code their. I'll create a branch on this repo with the code using `aws` and `aws` will be a dependency.

The maintainer of `aws` are willing to merge PRs making the code easier to re-use. If some parts of the code are really cloudflare specific or harder to integrate, we can copy/update the code and think later what the best to reconcile the code is.

> It talks about multiple possibilities being explored already with no information or context about those explorations or the outcomes, so unsure what the current state is

We only quickly brainstormed with Dario, Pete and Igor about possible ways to implement this. There is nothing decided yet. I guess having a first working implementation would help evaluate of feasibility and pros/cons of other possible solutions.

As mentioned on the cache issue there multiple parts to adapt. If you start working on something please create a dedicated issue and link to the main issue.

I hope this helps,
Thanks